### ISSUE TITLE
Initial feedback

### ISSUE THREAD

**Issue by thedarkjester** [7i7o/blockchain-developer-bootcamp-final-project#1]
You can close this issue after reading - it is more a note on the project.

I like the way the tests are laid out and that you linked the libraries - nicely done.

There are a few minor gas optimisations you could consider

you could adjust the two constants at the beginning to be 128 each and save a single storage slot. (RXStructs could be two slots vs. 4)

```
     uint256 constant MAX_KEY_LENGTH = 20;
    uint256 constant MAX_VALUE_LENGTH = 62;
    uint256 constant RX_LINES = 12; // Has to be same value as in RxStructs !!!
```
  
  Test out what happens to the gas on setting the values if you do the following by accessing the default object vs struct constructor. (same thing could be done with the pharmacist and subject removing the need for the 0 and blank strings)
  
  ```
  rxs[tokenId].PropertyNameHere = valueHere;
  ```
  instead of 
  ```
  rxs[tokenId] = RxStructs.RxData(
                                block.timestamp,
                                getSubject(to),
                                getSubject(msg.sender),
                                RxStructs.Subject(address(0), 0, '', ''),
                                getDoctor(msg.sender),
                                RxStructs.Pharmacist(address(0), '', ''),
                                _keys,
                                _values);
                                ```

**Comment by 7i7o** [comment]
I really appreciate you had the time to review and leave some feedback!

I'll be sure to implement these (and more) changes to save gas.

On top of the gas optimizing, I have a few more things in queue for the contract after grading (wan to override tokenURI() so the SVG is alive after burning).
I have a development branch where I added (and fixed) some stuff, but it is not included in the submission, so I have not merged yet.

Does this mean I passed or do I have to make some changes? 😊

Thanks again for taking the time to review and give me some useful feedback!
